### ISSUE TITLE
UserWarning: Implicit dimension choice for log_softmax has been deprecated. Change the call to include dim=X as an argument.   return F.log_softmax(x.view(-1, self.q_levels)) \

### ISSUE THREAD

**Issue by blibliki** [wekaco/samplernn-pytorch#14]
2) Pay attention to your initial loss.

Continuing the binary example, if your data is 30% 0's and 70% 1's, then your intial expected loss around 𝐿=−0.3ln(0.5)−0.7ln(0.5)≈0.7. This is because your model should start out close to randomly guessing.

A lot of times you'll see an initial loss of something ridiculous, like 6.5. Conceptually this means that your output is heavily saturated, for example toward 0. For example −0.3ln(0.99)−0.7ln(0.01)=3.2, so if you're seeing a loss that's bigger than 1, it's likely your model is very skewed. This usually happens when your neural network weights aren't properly balanced, especially closer to the softmax/sigmoid. So this would tell you if your initialization is bad.

You can study this further by making your model predict on a few thousand examples, and then histogramming the outputs. This is especially useful for checking that your data is correctly normalized. As an example, if you expect your output to be heavily skewed toward 0, it might be a good idea to transform your expected outputs (your training data) by taking the square roots of the expected output. This will avoid gradient issues for saturated sigmoids, at the output.
